custom_components/versatile_thermostat/underlyings.py

```diff
@@ -245,12 +255,13 @@ async def turn_off(self):
         domain = self._entity_id.split(".")[0]
         # This may fails if called after shutdown
         try:
-            data = {ATTR_ENTITY_ID: self._entity_id}
-            await self._hass.services.async_call(
-                domain,
-                command,
-                data,
-            )
+            try:
+                data = {ATTR_ENTITY_ID: self._entity_id}
+                await self._hass.services.async_call(domain, command, data)
+                self._keep_alive.set_async_action(self.turn_off)
+            except Exception:
+                self._keep_alive.cancel()
+                raise
```

Comment:
Building on previous comments, I’ve added an inner `try` block that catches and re-raises any exception for the purpose of cancelling keep alive if the service call fails for any reason. I had the following cases in mind:
- Keep alive initiated through the `startup()` method, before it is known whether the service calls will succeed.
- Keep alive initiated here, such that at least one service call succeeded, but where subsequent service calls might fail (for any unforeseen reason, perhaps in case the user renames or removes the service while Versatile Thermostat is running).

Adding the inner try/except block allows a single `cancel()` call to be made. An alternative might have been to make two `cancel()` calls:

```python
        try:
            data = {ATTR_ENTITY_ID: self._entity_id}
            await self._hass.services.async_call(domain, command, data)
            self._keep_alive.set_async_action(self.turn_on)
        except ServiceNotFound as err:
            _LOGGER.error(err)
            self._keep_alive.cancel()
        except Exception:
            self._keep_alive.cancel()
            raise
```

By the way:
- `pylint` doesn’t seem to complain about “broad exception caught” when the exception is re-raised, which is why I did not need to add `# pylint: disable=broad-exception-caught`.
- `black` was happy with the re-formatting.